eodatasets/prepare/ncep_reanalysis_surface_pr_wtr.py

```diff
@@ -0,0 +1,166 @@
+        }
+    }
+
+
+def get_uuid(collection, idx):
+    """
+    Returns a deterministic uuid based on band index and gdal checksum
+    """
+    origin = collection.tags()['NC_GLOBAL#References']
+
+    return uuid.uuid5(
+        NOAA_WATER_VAPOUR_NS,
+        '{}?{}'.format(origin, urllib.parse.urlencode({
+            'checksum': collection.checksum(idx),
+            'band_index': idx
+        }))
+    )
+
+
+def process_datasets(dataset: Path):
+    """
+    Generates a metadata document for each band available
+    in the water vapour file.
+
+    Each band is treated as a separate dataset since the
+    source file is updated in-place; allowing the file
+    to be modified but a static reference is provided for
+    each observation
+
+    """
+    datasets = []
+    creation_dt = (
+        datetime.datetime.utcnow()
+        .replace(tzinfo=datetime.timezone.utc)
+    )
+
+    with rasterio.open(str(dataset), 'r') as collection:
+        collection_start_date = (
+            datetime.datetime.strptime(
+                collection.tags()['time#units'],
+                'hours since %Y-%m-%d %H:%M:%S.%f'
+            )
+        )
+        creation_dt = (
+            datetime.datetime.utcnow()
+            .replace(tzinfo=datetime.timezone.utc)
+        )
+
+        for _idx in collection.indexes:
+            time_in_hours = int(collection.tags(_idx)['NETCDF_DIM_time'])
+            ds_dt = (
+                collection_start_date +
+                datetime.timedelta(hours=time_in_hours)
+            ).replace(tzinfo=datetime.timezone.utc)
+
+            md = {}
+            md['id'] = get_uuid(collection, _idx).hex
+            md['creation_dt'] = creation_dt.isoformat()
+            md['description'] = NOAA_DESCRIPTION
+            md['extent'] = {
+                'center_dt': ds_dt.isoformat(),
+                'coord': get_coords(collection)
+            }
+            md['grid_spatial'] = {
+                'projection': {
+                    'geo_ref_points': get_coords(collection, xkey='x', ykey='y'),
+                    'spatial_reference': rasterio.crs.CRS(init='EPSG:4236').wkt
+                }
+            }
+            md['format'] = {'name': 'netCDF'}
+            md['image'] = {
+                'bands': {
+                    'water_vapour': {
+                        'path': dataset.name,
+                        'layer': 'pr_wtr',
+                        'band': _idx
+                    }
+                }
+            }
+            md['product_type'] = 'auxiliary'
+            md['product_name'] = 'ncep_reanalysis_surface_pr_wtr'
+            md['sources'] = {}
+
+            datasets.append(md)
+
+    return datasets
+
+
+def _process_datasets(output_dir, datasets):
+    """
+    Wrapper function for processing multiple datasets
+    """
+    for dataset in datasets:
+        docs = process_datasets(dataset)
+        outfile = output_dir / (dataset.stem + '-metadata.yaml')
```

Comment:
We previously used `.ga-md.yaml` extension for sidecar metadata (there's a few codebases like sync which will check this: eg. https://github.com/GeoscienceAustralia/digitalearthau/blob/901a2988cadcdd91c4cbcc14105928b94dcfba09/digitalearthau/test_paths.py#L64